### PR TITLE
Update quarterly release checklist

### PR DESCRIPTION
**Pull request by sxa**
Draft as I want to evolve this as required during the October release cycle

Signed-off-by: Stewart X Addison <sxa@redhat.com>

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -7,12 +7,16 @@ assignees: ''
 
 ---
 
+**NOTE: Items marked `jdkxx` and `TEMPLATE_UPDATEME` should be replaced while deploying this issue template. It is recommented to delete this line once you've done so :-)**
```
Comment (smlambert) [inline .github/ISSUE_TEMPLATE/release-checklist.md:10]:
```suggestion
**NOTE: Items marked `jdkxx` and `TEMPLATE_UPDATEME` should be replaced while deploying this issue template. It is recommended to delete this line once you've done so :-)**
```

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 2]
```diff
@@ -47,24 +51,23 @@ Release Week Checklist:
...
 - [ ] **Verify binaries published successfully** to github releases repo and website (_automate_*, this could also be an automated test)
+- [ ] **Publish updates to the containers to dockerhub**
+- [ ] **Edit the [Homebrew Temurin Cask](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/temurin.rb) and replace the version and sha256 as appropriate.  This means for Homebrew users that they install the latest by default and can use the `@` notation to install older versions if they wish.
 - [ ] **Update support page** (_automate_* github workflow to create a PR to update [support.handlebars](https://github.com/adoptium/adoptium.net/blob/master/src/handlebars/support.handlebars))
 - [ ] **Update release notes** (_automate_* - github workflow to create update for [release notes page](https://adoptium.net/release_notes.html))
-- [ ] **Trigger linux installers pipeline** currently it is part of the build pipelines (will eventually be updated to run independently)
+- [ ] **Trigger linux installers pipeline** currently it is part of the b pipelines (will eventually be updated to run independently)
```
[the remark this answers is not in the document]
Comment (sxa) [reply .github/ISSUE_TEMPLATE/release-checklist.md]: No - I've already pushed another update correcting that :-)

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 1 above)
Comment (sxa) [reply .github/ISSUE_TEMPLATE/release-checklist.md]: Changed - thanks (For some reason it didn't let me directly commit your suggestion though)

**Review by karianna — APPROVED** [review]
LGTM

**Pull request merged**